girder/utility/s3_assetstore_adapter.py

```diff
@@ -155,10 +154,25 @@ def initUpload(self, upload):
     def uploadChunk(self, upload, chunk):
         """
         Rather than processing actual bytes of the chunk, this will generate
-        the signature required to upload the chunk.
+        the signature required to upload the chunk. Clients that do not support
+        direct-to-S3 upload can pass the chunk via the request body as with
+        other assetstores, and girder will proxy the data through to S3.
 
         :param chunk: This should be a JSON string containing the chunk number
-        and S3 upload ID.
+            and S3 upload ID. If a normal chunk file-like object is passed,
+            we will send the data to S3.
+        """
+        if isinstance(chunk, six.string_types):
+            return self._clientUploadChunk(upload, chunk)
+        else:
+            return self._proxiedUploadChunk(upload, chunk)
+
```

Comment:
Something more explicit than checking the type of the chunk would be ideal.  @zachmullen said something like #837 would help.